### DISCUSSION TITLE
PDF to SVG / Wrong SVG elements appearance

### DISCUSSION THREAD

**Discussion by serhii-brovarnyk** [pymupdf/PyMuPDF#3573] [Upstream Bugs] [answered]
Hello!

I have a PDF file with only one page I got via PyMuPDF from a multipage PDF doc, and I'm trying to get an SVG image of this PDF document page. The pixmap of this page has a completely different appearance from the SVG image I got via the `page.get_svg_image()` method.
There are big black-filled rectangles instead of transparent rectangular areas limited by corresponding rectangles without fill.

I've tried to convert this PDF doc via Inkscape and got the correct result so I think it is a bug on the PyMuPDF side.
The corresponding files (SVG files extension changed to .txt to prevent their uploading as images):
[test.pdf](https://github.com/user-attachments/files/15817322/test.pdf)
[test_inkscape.txt](https://github.com/user-attachments/files/15817344/test_inkscape.txt)
[test_pymupdf.txt](https://github.com/user-attachments/files/15817366/test_pymupdf.txt)

Thank you in advance for any help!

**Comment by JorjMcKie** [comment, marked as answer]
Created a MuPDF bug report here: https://bugs.ghostscript.com/show_bug.cgi?id=707826

**Reply by serhii-brovarnyk** [reply]
Thank you!
Do you have any ideas why this issue occurs? Maybe you have suggestions on how to solve it?
In the case of manually adding `fill="none"` to the SVG document header issue is solved (visually) but it should be wrong because of the wrong appearance of elements that should be filled in case of the absence of the `fill` parameter.

**Reply by JorjMcKie** [reply]
The actual code to generate SVG source is located in the base library MuPDF. So any insight or solution must come from that team - which I'm sure will happen in due time as always.

**Reply by serhii-brovarnyk** [reply]
Thank you for the answer and your help!

**Comment by JorjMcKie** [comment]
The MuPDF team already has responded that the cause was identified and a fix will be developed.

**Reply by serhii-brovarnyk** [reply]
It is wonderful!
Thank you very much!

**Reply by JorjMcKie** [reply]
Thanks for reporting this!
